aptos-move/e2e-move-tests/tests/stake.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_types::account_address::AccountAddress;
+use e2e_move_tests::{assert_success, enable_golden, MoveHarness};
+
+#[test]
+fn test_staking_end_to_end() {
+    let mut harness = MoveHarness::new();
+    enable_golden!(harness);
+    let owner = harness.new_account_at(AccountAddress::from_hex_literal("0x123").unwrap());
+    let operator = harness.new_account_at(AccountAddress::from_hex_literal("0x234").unwrap());
+    let owner_address = *owner.address();
+    let operator_address = *operator.address();
+
+    // Initialize and add stake.
+    let stake_amount = 100_000_000;
+    assert_success!(harness.initialize_staking(
+        &owner,
+        stake_amount,
+        operator_address,
+        owner_address
+    ));
+    let stake_pool = harness.get_stake_pool(&owner_address);
+    assert_eq!(stake_pool.active, stake_amount);
+    assert_eq!(stake_pool.operator_address, operator_address);
+    assert_eq!(stake_pool.delegated_voter, owner_address);
+
+    // Join the validator set.
+    assert_success!(harness.rotate_consensus_key(&operator, owner_address));
+    assert_success!(harness.join_validator_set(&operator, owner_address));
+    harness.new_epoch();
+
+    // Validator should now be locked up.
+    let stake_pool = harness.get_stake_pool(&owner_address);
+    assert_eq!(
+        stake_pool.locked_until_secs,
+        harness.executor.get_block_time_seconds() + 7200
+    );
+
+    // Unlock stake.
+    assert_success!(harness.unlock_stake(&owner, stake_amount / 2));
```

Comment:
let's have unlocked stake and stake left be different, so you can confirm right amount is removed. i.e. 1/4 and 3/4 for example?

Reply:
Updated.